control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp

```diff
@@ -111,53 +130,60 @@ VehicleCmdGate::VehicleCmdGate() : nh_(""), pnh_("~"), is_engaged_(false), is_em
   filter_.setLatJerkLim(lat_jerk_lim);
 
   // Set default value
-  current_gate_mode_.data = autoware_control_msgs::GateMode::AUTO;
+  current_gate_mode_.data = autoware_control_msgs::msg::GateMode::AUTO;
 
   // Timer
-  timer_ = pnh_.createTimer(ros::Rate(update_rate_), &VehicleCmdGate::onTimer, this);
+  auto timer_callback = std::bind(&VehicleCmdGate::onTimer, this);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(update_period_));
+  timer_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_callback)>>(
```

Comment:
@TakaHoribe @fred-apex-ai This is the timer replacement. I tried it out with a minimal publisher that sends `/clock` messages and a timer like this. The timer stops when no messages are sent and sim time is enabled, and goes faster when the published clock goes faster.

Reply:
Thank you! I'll test with this code in my package.

Reply:
@nnmm I will try it out, too